docs/development.md

```diff
@@ -0,0 +1,46 @@
+# Development
+
+## Prerequisites
+
+You must have a recent version of [bazel](https://bazel.io) installed. Bazel is
+the recommended way to build and test the cluster registry. Bazel is designed to
+maintain compatibility with standard Go tooling, but this is not tested on a
+regular basis, and some scripts/tooling in this project are built around Bazel.
+
+NOTE: There is an issue with bazel 0.6.x. As a workaround, use 0.5.x, or pass
+the flag `--incompatible_comprehension_variables_do_not_leak=false` to bazel
+0.6.x invocations.
+
+Before doing any development work, you must (in order, from the repository root
+directory, after cloning):
+
+1.  run `./update-codegen.sh`
+1.  run `bazel run //:gazelle`
+
+## Building `clusterregistry`
+
+`clusterregistry` is the binary for the Kubernetes API server that serves the
+cluster registry API.
+
+To build it, from the root of the repository:
+
+1.  Run `bazel build //cmd/clusterregistry`
```

Comment:
(note: this might take a while the first time you run it...)

Reply:
Good point. Added a note to that effect.